src/Appwrite/Platform/Workers/Functions.php

```diff
@@ -194,6 +194,70 @@ function: $function,
         }
     }
 
+    /**
+     * @param string $message
+     * @param Document $function
+     * @param string $trigger
+     * @param string $path
+     * @param string $method
+     * @param Document $user
+     * @param string|null $jwt
+     * @param string|null $event
+     * @throws Exception
+     */
+    private function fail(
+        string $message,
+        Database $dbForProject,
+        Document $function,
+        string $trigger,
+        string $path,
+        string $method,
+        Document $user,
+        string $jwt = null,
+        string $event = null,
+    ): void {
+        $headers['x-appwrite-trigger'] = $trigger;
+        $headers['x-appwrite-event'] = $event ?? '';
+        $headers['x-appwrite-user-id'] = $user->getId() ?? '';
+        $headers['x-appwrite-user-jwt'] = $jwt ?? '';
+
+        $headersFiltered = [];
+        foreach ($headers as $key => $value) {
+            if (\in_array(\strtolower($key), FUNCTION_ALLOWLIST_HEADERS_REQUEST)) {
+                $headersFiltered[] = ['name' => $key, 'value' => $value];
+            }
+        }
+
+        $executionId = ID::unique();
+        $execution = new Document([
+            '$id' => $executionId,
+            '$permissions' => $user->isEmpty() ? [] : [Permission::read(Role::user($user->getId()))],
+            'functionInternalId' => $function->getInternalId(),
+            'functionId' => $function->getId(),
+            'deploymentInternalId' => '',
+            'deploymentId' => '',
+            'trigger' => $trigger,
+            'status' => 'failed',
+            'responseStatusCode' => 400,
+            'responseHeaders' => [],
+            'requestPath' => $path,
+            'requestMethod' => $method,
+            'requestHeaders' => $headersFiltered,
+            'errors' => $message,
+            'logs' => '',
+            'duration' => 0.0,
```

Comment:
Can we still calculate duration? Its important because compute was still used.

Reply:
Even in the execute method, we start calculating the duration after the execution document is created. Since we throw the error just after creating the document and since we're not actually executing anything, the duration should be 0 in my opinion.